### PR TITLE
Fixing numpy fill diagonal test

### LINKED ISSUE #14585: Fix inserting_data_into_arrays.test_numpy_fill_diagonal
|   |   |
|---|---|
|paddle|<a href="https://github.com/unifyai/ivy/actions/runs/6219910913"></a>
|tensorflow|<a href="https://github.com/unifyai/ivy/actions/runs/6219910913"></a>
|torch|<a href="https://github.com/unifyai/ivy/actions/runs/6219910913"></a>
|numpy|<a href="https://github.com/unifyai/ivy/actions/runs/6219910913"></a>
|jax|<a href="https://github.com/unifyai/ivy/actions/runs/6219910913"></a>

### PR DESCRIPTION
**Pull request by abdulasiraj**
closes #14585

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_inserting_data_into_arrays.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
         min_dim_size=2,
         max_num_dims=2,
     ),
-    val=helpers.floats(),
+    val=helpers.floats(min_value=-10, max_value=10),
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_inserting_data_into_arrays.py:19]: We should not restrict it to such a small number,
We only restrict data generations in hypothesis for those functions which are numerically unstable on big numbers
This is just the value to be populated in the diagonal, try not restricting it and see if it passes the test.

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @abdulasiraj

Requested some changes :slightly_smiling_face: 

Do let me know if you any some questions!
Thanks! :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_inserting_data_into_arrays.py (hunk 1 above)
Comment (abdulasiraj) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_inserting_data_into_arrays.py:19]: Most of the time test are passed without restriction but occasionally some backends are unable to handle floating point precision. In my opinion, I think it's necessary to restrict floating point numbers. But I'm more than happy to know if there's any other way to handle this thing.

**Review by zaeemansari70 — APPROVED** [review]
If some backends are failing for the reason you stated i.e `some backends are unable to handle floating point precision`
Then the changes look good to me :+1: :slightly_smiling_face: 
The tests also seem to be passing :slightly_smiling_face:

**Pull request merged**